.github/CONTRIBUTING.md

```diff
@@ -25,49 +25,52 @@ In VDP, a **pipeline** is a DAG (Directed Acyclic Graph) consisting of multiple
 
 ```mermaid
 flowchart LR
-    s[Trigger] --> c1[OpenAI Connector]
-    c1 --> c2[Stability AI Connector]
-    c1 --> c3[MySQL Connector]
+    s[Trigger] --> c1[OpenAI Component]
+    c1 --> c2[Stability AI Component]
+    c1 --> c3[MySQL Component]
     c1 --> e[Response]
     c2 --> e
 ```
 
 ### Component
 
 There are different types of component:
-- **connector**
-  - Queries, processes or transmits the ingested data to a service or app.
-  - Users need to configure their connectors (e.g. by providing an API token to
-    a remote service).
+- **ai**
+  - Transforms unstructured data into formats that are easy to interpret and analyze, thereby facilitating the extraction of valuable insights.
+- **data**
+  - Establishes connections with remote data sources, such as IoT devices (e.g., IP cameras), cloud storage services (e.g., GCP Cloud Storage, AWS S3), data warehouses, or vector databases (e.g., Pinecone).
+- **application**
+  - Integrates various 3rd-party application services
 - **operator**
   - Performs data injection and manipulation.
 - **iterator**
   - Takes an array and executes an operation (defined by a set of nested
     components) on each of its elements.
-- **trigger / response**
-  - These special components provide an input / output interface to pipeline
-    triggers.
 
-#### Connector
+#### AI
 
-- **Connectors** are used by the pipeline to interact with an external service.
-  They are defined and initialized in the [connector](../pkg/connector) package.
-- In order to set up a connector, you may need to introduce its **connection**
-  details in the connector properties.
-  - In order to prevent private keys from being unintentionally leaked when
-    sharing a pipeline, the connection properties only take reference to a
-    **secret** (e.g. `${secrets.my-secret}`).
-  - You can create secrets from the console settings or through an [API
-    call](https://openapi.instill.tech/reference/pipelinepublicservice_createusersecret).
+- **AI** components play a crucial role in transforming unstructured data into formats that are easy to interpret and analyze, thereby facilitating the extraction of valuable insights. These components integrate with AI models from various providers, whether it's the primary Instill Model or those from third-party AI vendors. They are defined and initialized in the [ai](../ai) package.
```

Comment:
I'd remove the bullets in these sections, as there's one single point and no sub-bullets. In `Connection`, we can keep the 2nd level of bullets, outdented.
```suggestion
**AI** components play a crucial role in transforming unstructured data into formats that are easy to interpret and analyze, thereby facilitating the extraction of valuable insights. These components integrate with AI models from various providers, whether it's the primary Instill Model or those from third-party AI vendors. They are defined and initialized in the [ai](../ai) package.
```

Reply:
It looks weird that `Connection` has the same level as components, I use blockquote for the connection notes, what do you think?

> **Note:**
> - For **AI**, **Data**, **Application** components, they are used by the pipeline to interact with an external service, you may need to introduce its **connection** details in the component connection properties.
>   - In order to prevent private keys from being unintentionally leaked when
    sharing a pipeline, the connection properties only take reference to a
    **secret** (e.g. `${secrets.my-secret}`).
>   - You can create secrets from the console settings or through an [API
    call](https://openapi.instill.tech/reference/pipelinepublicservice_createusersecret).

Reply:
Yeah I thought the same. I'd put it before the `#### AI` section, though. Otherwise it looks like it's part of the Iterator block.

Reply:
Good suggestion, let me move it.

Reply:
moved